src/libopenrave/robotconnectedbody.cpp

```diff
@@ -424,6 +424,35 @@ void RobotBase::ConnectedBody::ExtractInfo(RobotBase::ConnectedBodyInfo& info) c
 {
     // TODO: extract info from connectedbody to make extra geometry working
     info = _info;
+
+    for(size_t iLinkInfo = 0; iLinkInfo < _info._vLinkInfos.size(); ++iLinkInfo) {
+        info._vLinkInfos[iLinkInfo].reset(new KinBody::LinkInfo());
```

Comment:
`info._vLinkInfos[iLinkInfo] = boost::make_shared<KinBody::LinkInfo>()` to avoid memory leak when exception is thrown.